<?xml version="1.0"?>
<launch>
  <arg name="rosbag_path" default="/home/nvidia/Downloads/static_avoidance.bag" />
  <arg name="rviz_config_path" default="$(find static_avoidance)/rviz_config/static_avoidance.rviz" />

  <node pkg="static_avoidance" name="static_avoidance_node" type="static_avoidance_node" output="screen">
	  <param name="DETECT_DISTANCE" type="int" value="4"/>
	  <param name="CONST_STEER" type="int" value="0"/>
	  <param name="CONST_VEL" type="int" value="6"/>
	  <param name="OBSTACLE_RADIUS" type="float" value="0.3"/>
	  <param name="TURN_FACTOR" type="float" value="2.5"/>
	  <param name="TURN_WEIGHT" type="int" value="10"/>
	  <param name="RETURN_WEIGHT" type="int" value="12"/>
  </node>

  <node pkg="rosbag" type="play" name="play" args="-l $(arg rosbag_path)"/> 

  <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="laser_filter">
    <rosparam command="load" file="$(find laser_filters)/examples/box_filter.yaml" />
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_config_path)"/>

  <node name="obstacle_extractor" pkg="obstacle_detector" type="obstacle_extractor_node">
	<remap from="scan" to="scan_filtered"/>
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>

    <param name="use_split_and_merge"    value="true"/>
    <param name="circles_from_visibles"  value="true"/>
    <param name="discard_converted_segments" value="true"/>
    <param name="transform_coordinates"  value="true"/>

    <param name="min_group_points"     value="5"/>

    <param name="max_group_distance"   value="0.1"/>
    <param name="distance_proportion"  value="0.00628"/>
    <param name="max_split_distance"   value="0.2"/>
    <param name="max_merge_separation" value="0.2"/>
    <param name="max_merge_spread"     value="0.2"/>
    <param name="max_circle_radius"    value="1.0"/>
    <param name="radius_enlargement"   value="0.3"/>

    <param name="frame_id"             value="laser"/>
  </node>

</launch>
